engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -1232,50 +1245,55 @@ namespace dmGameSystem
             //   | |  w  | |
             // 3 *-*-----*-*
-            float us[4], vs[4], xs[4], ys[4];
 
             // v are '1-v'
             xs[0] = ys[0] = 0;
             xs[3] = ys[3] = 1;
-
-            // disable slice9 computation below a certain dimension
-            // (avoid div by zero)
-            const float s9_min_dim = 0.001f;
-
-            const float su = 1.0f / org_width;
-            const float sv = 1.0f / org_height;
-            const float sx = size.getX() > s9_min_dim ? 1.0f / size.getX() : 0;
-            const float sy = size.getY() > s9_min_dim ? 1.0f / size.getY() : 0;
-
-            static const uint32_t uvIndex[2][4] = {{0,1,2,3}, {3,2,1,0}};
-            bool uv_rotated = tc[0] != tc[2] && tc[3] != tc[5];
-            bool flip_u, flip_v;
-            GetNodeFlipbookAnimUVFlip(scene, node, flip_u, flip_v);
-            if(uv_rotated)
+            bool uv_rotated;
+            if(tc)
             {
-                const uint32_t *uI = flip_v ? uvIndex[1] : uvIndex[0];
-                const uint32_t *vI = flip_u ? uvIndex[1] : uvIndex[0];
-                us[uI[0]] = tc[0];
-                us[uI[1]] = tc[0] + (su * slice9.getW());
-                us[uI[2]] = tc[2] - (su * slice9.getY());
-                us[uI[3]] = tc[2];
-                vs[vI[0]] = tc[1];
-                vs[vI[1]] = tc[1] - (sv * slice9.getX());
-                vs[vI[2]] = tc[5] + (sv * slice9.getZ());
-                vs[vI[3]] = tc[5];
+                static const uint32_t uvIndex[2][4] = {{0,1,2,3}, {3,2,1,0}};
+                uv_rotated = tc[0] != tc[2] && tc[3] != tc[5];
+                bool flip_u, flip_v;
+                GetNodeFlipbookAnimUVFlip(scene, node, flip_u, flip_v);
+                if(uv_rotated)
+                {
+                    const uint32_t *uI = flip_v ? uvIndex[1] : uvIndex[0];
+                    const uint32_t *vI = flip_u ? uvIndex[1] : uvIndex[0];
+                    us[uI[0]] = tc[0];
+                    us[uI[1]] = tc[0] + (su * slice9.getW());
+                    us[uI[2]] = tc[2] - (su * slice9.getY());
+                    us[uI[3]] = tc[2];
+                    vs[vI[0]] = tc[1];
+                    vs[vI[1]] = tc[1] - (sv * slice9.getX());
+                    vs[vI[2]] = tc[5] + (sv * slice9.getZ());
+                    vs[vI[3]] = tc[5];
+                }
+                else
+                {
+                    const uint32_t *uI = flip_u ? uvIndex[1] : uvIndex[0];
+                    const uint32_t *vI = flip_v ? uvIndex[1] : uvIndex[0];
+                    us[uI[0]] = tc[0];
+                    us[uI[1]] = tc[0] + (su * slice9.getX());
+                    us[uI[2]] = tc[4] - (su * slice9.getZ());
+                    us[uI[3]] = tc[4];
+                    vs[vI[0]] = tc[1];
+                    vs[vI[1]] = tc[1] + (sv * slice9.getW());
+                    vs[vI[2]] = tc[3] - (sv * slice9.getY());
+                    vs[vI[3]] = tc[3];
+                }
```

Comment:
It's unclear what the bugfix is?
This code looks the same as before?

Reply:
Answered: https://github.com/defold/defold/pull/4921#issuecomment-644750126